pkgs/build-support/setup-hooks/die.sh

```diff
@@ -0,0 +1,21 @@
+# Exit with backtrace and error message
+#
+# Usage: die "Error message"
+die() {
+    # Let us be a little sloppy with errors, because otherwise the final
+    # invocation of `caller` below will cause the script to exit.
+    set +e
+
+    # Print our error message
+    printf "\nBuilder called die: %b\n" "$*"
+    printf "Backtrace:\n"
+
+    # Print a backtrace.
+    local frame=0
+    while caller $frame; do
+        ((frame++));
+    done
+    printf "\n"
+
+    exit 1
+}
```

Comment:
It varies in usefulness, depending on how makeWrapper is being called.  Sometimes there's enough information to be helpful, but often the stack trace stops at the point where `stdenv/setup.sh` evaluates the current phase hook.